### PR TITLE
C# method snippets

### PR DESCRIPTION
**Pull request by chrisdunelm**

### CODE REVIEW

File: src/test/java/com/google/api/codegen/testdata/csharp_wrapper_samples_library.baseline [hunk 1]
```diff
@@ -0,0 +1,494 @@
...
+
+    public class LibraryServiceClientSnippets
+    {
+        public async Task CreateShelfAsync()
+        {
+            // Snippet: CreateShelfAsync(Shelf,CallSettings)
+            // Additional: CreateShelfAsync(Shelf,CancellationToken)
+            LibraryServiceClient client = await LibraryServiceClient.CreateAsync();
+            Shelf shelf = new Shelf();
+            Shelf response = await client.CreateShelfAsync(
```
Comment (jskeet) [inline src/test/java/com/google/api/codegen/testdata/csharp_wrapper_samples_library.baseline]: Unwrap parameters if they fit within some boundary? This looks odd at the moment.

File: src/test/java/com/google/api/codegen/testdata/csharp_wrapper_samples_library.baseline [hunk 2]
```diff
@@ -0,0 +1,494 @@
...
+                shelf);
+            // End snippet
+        }
+
+        public async Task GetShelfAsync()
+        {
+            // Snippet: GetShelfAsync(string,CallSettings)
+            // Additional: GetShelfAsync(string,CancellationToken)
+            LibraryServiceClient client = await LibraryServiceClient.CreateAsync();
+            string name = "";
```
Comment (jskeet) [inline src/test/java/com/google/api/codegen/testdata/csharp_wrapper_samples_library.baseline]: Hopefully we'll give better values than this at some point?

File: src/test/java/com/google/api/codegen/testdata/csharp_wrapper_samples_library.baseline [hunk 3]
```diff
@@ -0,0 +1,494 @@
...
+            // Additional: GetShelfAsync(string,SomeMessage,CancellationToken)
+            LibraryServiceClient client = await LibraryServiceClient.CreateAsync();
+            string name = "";
+            SomeMessage message = new SomeMessage();
+            Shelf response = await client.GetShelfAsync(
+                name, message);
+            // End snippet
+        }
+
+        public void GetShelf()
```
Comment (jskeet) [inline src/test/java/com/google/api/codegen/testdata/csharp_wrapper_samples_library.baseline]: We appear to have two GetShelf and two GetShelfAsync methods (in the snippets file). Presumably this is due to multiple parameter flattenings... you could either just label them with a number, or put the parameters in the method name.

File: src/test/java/com/google/api/codegen/testdata/csharp_wrapper_samples_library.baseline [hunk 4]
```diff
@@ -0,0 +1,494 @@
...
+        }
+
+        public void ListShelves()
+        {
+            // Snippet: ListShelves(string,int?,CallSettings)
+            LibraryServiceClient client = LibraryServiceClient.Create();
+            string pageToken = null;
+            int? pageSize = null;
+            IPagedEnumerable<ListShelvesResponse, Shelf> response = client.ListShelves(
+                pageToken, pageSize);
```
Comment (jskeet) [inline src/test/java/com/google/api/codegen/testdata/csharp_wrapper_samples_library.baseline]: Can we show an example of iterating, both here and in the async version?

File: src/test/java/com/google/api/codegen/testdata/csharp_wrapper_samples_library.baseline [hunk 5]
```diff
@@ -0,0 +1,494 @@
...
+        }
+
+        public async Task ListShelvesAsync()
+        {
+            // Snippet: ListShelvesAsync(string,int?,CallSettings)
+            // Additional: ListShelvesAsync(string,int?,CancellationToken)
+            LibraryServiceClient client = await LibraryServiceClient.CreateAsync();
+            string pageToken = null;
+            int? pageSize = null;
+            IPagedEnumerable<ListShelvesResponse, Shelf> response = await client.ListShelvesAsync(
```
Comment (jskeet) [inline src/test/java/com/google/api/codegen/testdata/csharp_wrapper_samples_library.baseline]: This should be IPagedAsyncEnumerable, I think.

File: src/test/java/com/google/api/codegen/testdata/csharp_wrapper_samples_library.baseline [hunk 6]
```diff
@@ -0,0 +1,494 @@
...
+                name);
+            // End snippet
+        }
+
+        public void DeleteShelf()
+        {
+            // Snippet: DeleteShelf(string,CallSettings)
+            LibraryServiceClient client = LibraryServiceClient.Create();
+            string name = "";
+            void response = client.DeleteShelf(
```
Comment (jskeet) [inline src/test/java/com/google/api/codegen/testdata/csharp_wrapper_samples_library.baseline]: That's an interesting local variable type...

File: src/test/java/com/google/api/codegen/testdata/csharp_wrapper_samples_library.baseline (hunk 2 above)
Comment (garrettjonesgoogle) [inline src/test/java/com/google/api/codegen/testdata/csharp_wrapper_samples_library.baseline]: This implementation is not taking advantage of the work we have done to make more complete initialization of values for sample code.

**Comment by jskeet** [comment]
Haven't looked at the implementation at all yet. Just first feedback on the generated code. When it's looking more likely to compile, I'll pull it and see what else is missing.

**Pull request closed without merging**